test/test_pytools.py

```diff
@@ -362,6 +362,81 @@ def test_make_obj_array_iteration():
 # }}}
 
 
+# {{{ test obj array vectorization and decorators
+
+def test_obj_array_vectorize(c=1):
+    np = pytest.importorskip("numpy")
+    la = pytest.importorskip("numpy.linalg")
+
+    # {{{ functions
+
+    import pytools.obj_array as obj
+
+    def add_one(ary):
+        assert ary.dtype.char != "O"
+        return ary + c
+
+    def two_add_one(x, y):
+        assert x.dtype.char != "O" and y.dtype.char != "O"
+        return x * y + c
+
+    @obj.obj_array_vectorized
+    def vectorized_add_one(ary):
+        assert ary.dtype.char != "O"
+        return ary + c
+
+    @obj.obj_array_vectorized_n_args
+    def vectorized_two_add_one(x, y):
+        assert x.dtype.char != "O" and y.dtype.char != "O"
+        return x * y + c
+
+    class Adder:
+        def __init__(self, c):
+            self.c = c
+
+        @obj.obj_array_vectorized_n_args
```

Comment:
Adding this as a decorator on a class doesn't actually work anymore.

From looking at the code in `decorator`, they do some fancy stuff with `inspect` to get `self` back into the arguments. I'll look into fixing that too.

Reply:
236e746?

Reply:
Wow. I just spent a long time understanding what's going on here, as evidenced by the blurb I added above. TIL about a less-than-pretty corner of Python. Long story short: What you did LGTM. :)

Reply:
I'm still not sure I understand it properly!